akka-actor/src/main/scala/akka/util/ByteString.scala

```diff
@@ -364,21 +364,57 @@ object ByteString {
   }
 
   /**
+   * This class helps to optimise the access time for linear access such as looping all items of [[ByteStrings]].
+   *
+   * @param lastIndex the index of the last access
+   * @param arrayIndex the array index of the last access
+   * @param seen the amount of elements being seen, excluding the elements in current array
+   */
+  private final case class LinearScanHint(lastIndex: Int, arrayIndex: Int, seen: Int)
```

Comment:
> need more extra space.

Where?

> have to loop from starting point when the index moves to the next array.

This can be fixed:
```scala
class LinearScanHint(var pos: Int, var offset: Int)

val _hint = LinearScanHint(bytestrings(0), 0)

def apply(idx: Int): Byte = {
  while (_hint.offset > idx || idx >= _hint.offset + bytestrings(_hint.pos).length) {
    if (pos == bytestrings.length - 1) {
      _hint.pos = 0
      _hint.offset = 0
    } else {
      _hint.pos += 1
      _hint.offset += bytestrings(_hint.pos).length
    }
    bytestrings(_hint.pos)(idx - _hint.offset)
  }
}
```